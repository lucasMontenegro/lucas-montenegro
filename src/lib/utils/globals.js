import g from "global"
// renaming "global" to "globals"
// declaring a variable named "global" may cause trouble
export default g