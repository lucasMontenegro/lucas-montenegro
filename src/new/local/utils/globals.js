// renaming "global" to "globals"
// declaring a variable named "global" may cause troubles
exports.default = require("global")