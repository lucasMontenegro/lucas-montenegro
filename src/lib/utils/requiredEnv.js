import globals from "lib/utils/globals"
export default function requiredEnv (key) {
  const { env } = globals.process
  if (key in env) {
    return env[key]
  }
  throw Error(`process.env.${key} is required`)
}