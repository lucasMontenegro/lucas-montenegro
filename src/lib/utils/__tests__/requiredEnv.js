import globals from "lib/utils/globals"
import requiredEnv from "../requiredEnv"
jest.mock(`lib/utils/globals`, () => ({
  __esModule: true,
  default: { process: {} },
}))
describe(`../requiredEnv`, () => {
  jestUtils.describeDependencies({
    deps: [`lib/utils/globals`],
    relativeBasePath: __dirname,
  })
  describe(`requiredEnv (variable is defined)`, () => {
    it(`should return the environment value`, () => {
      globals.process.env = { FOO: `foo` }
      expect(requiredEnv(`FOO`)).toBe(`foo`)
    })
  })
  describe(`requiredEnv (variable is not defined)`, () => {
    it(`should throw`, () => {
      globals.process.env = {}
      expect(() => requiredEnv(`FOO`)).toThrow(`process.env.FOO is required`)
    })
  })
})