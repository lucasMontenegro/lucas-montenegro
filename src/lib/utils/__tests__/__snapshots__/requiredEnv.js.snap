// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../requiredEnv local dependencies lib/utils/globals should match the saved hash 1`] = `"658dffcc9a7d7107301d9ee254ba487ada2e2497635e046c509db0828ff3533f"`;
